lib/cppcheck.cpp

```diff
@@ -123,7 +123,15 @@ unsigned int CppCheck::checkFile(const std::string& filename, const std::string
         mErrorLogger.reportOut(std::string("Checking ") + fixedpath + ' ' + cfgname + std::string("..."));
 
         if (mSettings.verbose) {
-            mErrorLogger.reportOut("Defines: " + mSettings.userDefines);
+            mErrorLogger.reportOut("Defines:" + mSettings.userDefines);
+            std::string undefs;
+            for (const std::string& U : mSettings.userUndefs)
+            {
+                if (!undefs.empty())
+                    undefs += ";";
```

Comment:
please use `';'` instead of `";"`. It is less complex to append a char than a zero-terminated-string.